lib/request_handler/schema_parser.rb

```diff
@@ -14,18 +14,18 @@ def initialize(schema:, schema_options: {})
 
     private
 
-    def validate_schema(data)
+    def validate_schema(data, with: schema)
       raise MissingArgumentError, data: 'is missing' if data.nil?
-      validator = validate(data)
+      validator = validate(data, with: with)
       validation_failure?(validator)
       validator.output
     end
 
-    def validate(data)
+    def validate(data, with:)
```

Comment:
I think naming wise it would be more clear if we call the `with` parameter `schema` or `validation_schema`. Especially the else case below read weird `with.with`